src/dialects/mssql/query.js

```diff
@@ -41,6 +43,13 @@ export class MsSqlQuery extends AbstractQuery {
         // Default to a reasonable numeric precision/scale pending more sophisticated logic
         paramType.typeOptions = { precision: 30, scale: getScale(value) };
       }
+    } else if (typeof value === 'bigint') {
+      if (value < minSafeIntegerAsBigInt || value > maxSafeIntegerAsBigInt) {
+        paramType.type = TYPES.VarChar;
+        paramType.value = value.toString();
+      } else {
+        return this.getSQLTypeFromJsType(parseInt(value, 10), TYPES);
```

Comment:
I would prefer to use `Number()` to cast bigint to number, as `parseInt` first stringifies bigint then parses it using its weird logic :)

```suggestion
        return this.getSQLTypeFromJsType(Number(value), TYPES);
```

Reply:
Ah, I wasn't aware of the difference, thanks for pointing that out!